File: GraphicOverlayDemo/app/src/main/java/edu/cs4730/graphicoverlaydemo/MainActivity.java
package edu.cs4730.graphicoverlaydemo;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import androidx.annotation.NonNull;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.FragmentManager;
import androidx.appcompat.app.AppCompatActivity;

import edu.cs4730.graphicoverlaydemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {


    String TAG = "MainActivity";
    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        ViewCompat.setOnApplyWindowInsetsListener(binding.main, (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return WindowInsetsCompat.CONSUMED;
        });


        // if not using arch navigation, then you need to implement this.
        binding.navView.setOnItemSelectedListener(
            new BottomNavigationView.OnItemSelectedListener() {
                public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                    //setup the fragments here.
                    int id = item.getItemId();
                    if (id == R.id.nav_camera) {
                        getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new CameraFragment()).commit();

                    } else if (id == R.id.nav_simple) {
                        getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new SimpleFragment()).commit();
                    } else if (id == R.id.nav_multi) {
                        getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new MultiFragment()).commit();
                    }
                    item.setChecked(true);
                    return true;
                }
            }

        );
        //set the first one in place.
        getSupportFragmentManager().beginTransaction().add(binding.container.getId(), new SimpleFragment()).commit();
    }


}

File: GraphicOverlayDemo_kt/app/src/main/java/edu/cs4730/graphicoverlaydemo_kt/MainActivity.kt
package edu.cs4730.graphicoverlaydemo_kt

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import edu.cs4730.graphicoverlaydemo_kt.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    var TAG = "MainActivity"
    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        // if not using arch navigation, then you need to implement this.
        binding.navView.setOnItemSelectedListener { item -> //setup the fragments here.
            val id = item.itemId
            if (id == R.id.nav_camera) {
                supportFragmentManager.beginTransaction().replace(binding.container.id, CameraFragment())
                    .commit()
            } else if (id == R.id.nav_simple) {
                supportFragmentManager.beginTransaction().replace(binding.container.id, SimpleFragment())
                    .commit()
            } else if (id == R.id.nav_multi) {
                supportFragmentManager.beginTransaction().replace(binding.container.id, MultiFragment())
                    .commit()
            }
            item.isChecked = true
            true
        }
        //set the first one in place.
        supportFragmentManager.beginTransaction().add(binding.container.id, SimpleFragment()).commit()
    }

}
